compiler/rustc_trait_selection/src/solve/eval_ctxt/mod.rs

```diff
@@ -95,34 +95,7 @@ pub struct EvalCtxt<'a, 'tcx> {
     // evaluation code.
     tainted: Result<(), NoSolution>,
 
-    pub(super) inspect: ProofTreeBuilder<'tcx>,
-}
-
-#[derive(Default, Debug, Clone)]
-pub(super) struct NestedGoals<'tcx> {
```

Comment:
we would keep them in the actual impl crate for the trait solver, not the `ir` crate? I would maybe prefer to change this to `NestedGoals<I: Interner>` but keep it in this crate. And then once we've finished abstracting the solver move everything at once. That doesn't require us to change the visibility of these types